src/hotspot/share/adlc/output_c.cpp

```diff
@@ -3127,6 +3127,9 @@ static void define_fill_new_machnode(bool used, FILE *fp_cpp) {
     fprintf(fp_cpp, "    if( i != cisc_operand() ) \n");
     fprintf(fp_cpp, "      to[i] = _opnds[i]->clone();\n");
     fprintf(fp_cpp, "  }\n");
+    fprintf(fp_cpp, "  // Do not increment node index counter, since node reuses my index\n");
+    fprintf(fp_cpp, "  Compile* C = Compile::current();\n");
+    fprintf(fp_cpp, "  C->set_unique(C->unique() - 1);\n");
```

Comment:
This code seems to be static with no specializations.  It would be nice if it could be in a regular .cpp file. The only complication seems to be including it only if `used` is set, which could translate into an #ifdef.

Reply:
Or maybe something as simple as:
```
if (used) {
  fprintf(fp_cpp, "#include \"machnode.cpp.inc\"");
}
```

Reply:
Thanks for the suggestion, Dean. I am also a bit doubtful about having all this static code in `output_c.cpp`, my suggestion would be simply moving the entire definition of `fill_new_machnode()` to `machnode.cpp` (as in https://github.com/openjdk/jdk/commit/7f5c739919b166354d96e190702a515bbd3b3f5a). But I think it is better to address this (pre-existing) issue in a separate RFE and keep this bug fix small. What do you think?

Reply:
A separate RFE is fine.  Thanks.

Reply:
Thanks, reported in [JDK-8318959](https://bugs.openjdk.org/browse/JDK-8318959). Will do after integrating this changeset.